.github/workflows/build_portable_linux_artifacts.yml

```diff
@@ -118,10 +120,27 @@ jobs:
           extra_cmake_options: ${{ inputs.extra_cmake_options }}
           BUILD_DIR: build
         run: |
-          python3 build_tools/github_actions/build_configure.py --manylinux
+          python3 build_tools/github_actions/build_configure.py
 
       - name: Build therock-archives and therock-dist
-        run: cmake --build build --target therock-archives therock-dist -- -k 0
+        run: |
+          if [[ ${{ inputs.build_variant_label }} == "ASAN" ]]; then
+            # Background telemetry loop
+            ( while true; do
+                echo "--- $(date) ---"
+                free -h
+                top -b -n1 | head -20
+                sleep 30
+              done ) &
+            TELEMETRY_PID=$!
+          fi
+
+          cmake --build build --target therock-archives therock-dist -- -k 0
+
+          if [[ ${{ inputs.build_variant_label }} == "ASAN" ]]; then
+            # Stop telemetry loop
+            kill $TELEMETRY_PID
+          fi
```

Comment:
Let's add telemetry in a separate pull request. This has a few design issues worth exploring on their own.

* No inlined bash in workflow files, particularly surrounding critical path code like the CMake build (as it obscures the most relevant lines of the file).
* We can put code in the CMake build, a wrapper build script, a github workflow, on the runner itself, etc. There are tradeoffs with all approaches.